packages/wrangler/src/config/index.ts

```diff
@@ -32,17 +33,31 @@ export function readConfig<CommandArgs>(
 		Pick<OnlyCamelCase<CommonYargsOptions>, "experimentalJsonConfig">
 ): Config {
 	let rawConfig: RawConfig = {};
+	let isPagesConfig = false;
+
 	if (!configPath) {
 		configPath = findWranglerToml(process.cwd(), args.experimentalJsonConfig);
 	}
+
 	// Load the configuration from disk if available
 	if (configPath?.endsWith("toml")) {
 		rawConfig = parseTOML(readFileSync(configPath), configPath);
+		/**
+		 * Check if configuration file belongs to a Pages project.
+		 *
+		 * The `pages_build_output_dir` config key is used to determine if the
+		 * configuration file belongs to a Workers or Pages project. This key
+		 * should always be set for Pages but never for Workers. Furthermore,
+		 * Pages projects currently have support only for `wrangler.toml`,
+		 * which is why we only perform the check in this code path.
+		 */
+		isPagesConfig = rawConfig.pages_build_output_dir?.trim() !== "";
```

Comment:
Is there a reason we can't just check for `rawConfig.pages_build_output_dir !== undefined`

Reply:
I went and thought abt this some more and I _think_ there actually isn't. 

It seems that both `pages dev` and `pages deploy` are fine if we pass in `" "` as the directory arg, which is the equivalent of `pages_build_output_dir` being an empty string in this validation logic. What in effect this means, is that the build output directory will be resolved to the (most prolly root) directory that contains the `wrangler.toml` file. Since the build output directory contains static assets + the built functions + all other metadata files, like `_routes.json`, _headers`, etc, this'll mean that all these files + `wrangler.toml` + potentially the `functions` directory, will all live under this (root) dir (unless I have an error in judgement here). This would be a pretty, IMHO, funky setup, given that on `pages deploy` we deploy the assets inside the build output dir, which in this particular case would mean everything I mentioned earlier would get deployed?

If we were to decide against supporting a use case such as 👆, then we should perform that check against an empty string. However, seeing as we currently have nothing in place to prevent that, we can let it slide

Reply:
Is the reason this is in here because we are explicitly not supporting wrangler.json for Pages projects?

Reply:
that's correct yes. I mention that in the comment as well

Reply:
Doh! I didn't read that comment fully.
Should we error if there is a wrangler.json with this field in it?

Reply:
oh, that's a very good callout! we totes should! I'll fix that!